lib/versioning/Version.js

```diff
@@ -0,0 +1,168 @@
+'use strict'; // eslint-disable-line strict
+
+/**
+ * Class for manipulating an object version.
+ * The format of a version: { isNull, isDeleteMarker, versionId, otherInfo }
+ */
+class Version {
+    /**
+     * Create a new version instantiation from its data object.
+     * @param {object} version - the data object to instantiate
+     * @param {boolean} version.isNull - is a null version
+     * @param {boolean} version.isDeleteMarker - is a delete marker
+     * @param {string} version.versionId - the version id
+     * @constructor
+     */
+    constructor(version) {
+        this.version = version || {};
+    }
+
+    /**
+     * Parse the version information from a string.
+     *
+     * @param {string} value - the string to parse
+     * @return {Version} - the version deserialized from the input string
+     */
+    static from(value) {
+        return new Version(value ? JSON.parse(value) : undefined);
+    }
+
+    /**
+     * [MetaData Internal] Check if a version is a place holder for deletion.
+     *
+     * @param {string} value - version to check
+     * @return {boolean} - whether this is a PHD version
+     */
+    static isPHD(value) {
+        // check the length of the value before parsing
+        // all PHD versions should have the same length
+        if (value.indexOf('isPHD') < 0) {
+            return false;
+        }
+        // parse the value if it has the length of a PHD version
+        try {
+            return Version.from(value).isPHDVersion();
+        } catch (exception) { // eslint-disable-line strict
+            return false; // nice, Vault
```

Comment:
Vault is sending plain text data not in JSON format. Vault should never contain "isPHD" in the data it sends, but if it does, we have this covered.